qiita_pet/templates/sitebase.html

```diff
@@ -244,6 +244,11 @@
               </ul>
             </li>
           </ul>
+          <ul class="nav navbar-nav">
+            <li>
+              <a href="https://github.com/biocore/qiita/blob/master/README.rst#current-features">Current and Future Features</a>
```

Comment:
The idea is to have it as visible as possible for users. Thus, a separate direct link.